### PR TITLE
Create route53 resource record, setup route53 package, change license

### PR DESCRIPTION
**Pull request by fishnix**
* Changes the license to AGPLv2
* Sets up the route53 package/service including error code handling
* Creates route53 resource records
* Adds a step in the website handler to create an Alias resource record for the cloudfront distribution
* Adds a ListDistributions endpoint and tests for the distribution actions for cloudfront

### CODE REVIEW

**Comment by damnski** [comment]
:+1:

File: route53/route53.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+	r := Route53{}
+	log.Infof("creating new session with key id %s in region %s", account.Akid, account.Region)
+	sess := session.Must(session.NewSession(&aws.Config{
+		Credentials: credentials.NewStaticCredentials(account.Akid, account.Secret, ""),
+		Region:      aws.String(account.Region),
+	}))
+	r.Service = route53.New(sess)
+	r.Domains = account.Domains
+	return r
+}
```
Comment (damnski) [inline route53/route53.go]: hmmm

File: cloudfront/distribution_test.go [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+	} else {
+		t.Errorf("expected apierror.Error, got: %s", reflect.TypeOf(err).String())
+	}
+
+	// * ErrCodeNoSuchFieldLevelEncryptionProfile "NoSuchFieldLevelEncryptionProfile"
+	// The specified profile for field-level encryption doesn't exist.
+	c.Service.(*mockCloudFrontClient).err = awserr.New(cloudfront.ErrCodeNoSuchFieldLevelEncryptionProfile, "InvalidArgument", nil)
+	_, err = c.CreateDistribution(context.TODO(), distConfig, tags)
+	if aerr, ok := err.(apierror.Error); ok {
+		if aerr.Code != apierror.ErrNotFound {
```
[the remark this answers is not in the document]
Comment (fishnix) [reply cloudfront/distribution_test.go]: i agree with you, but i think this should be done more holistically now that I'm looking at it. i think ill leave this for now, but we should revisit soon in a separate PR for all of the types that we're dealing with.

File: common/config.go [hunk 3]
```diff
@@ -36,7 +36,8 @@ type AccessLog struct {
 
 // Domain is the domain configuration for an S3 site
 type Domain struct {
-	CertArn string
+	CertArn      string
+	HostedZoneID string
```
[the remark this answers is not in the document]
Comment (fishnix) [reply common/config.go]: 👍

File: cloudfront/distribution_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (tenyo) [reply cloudfront/distribution_test.go]: 👍

File: README.md [hunk 4]
```diff
@@ -456,6 +456,97 @@ POST `/v1/s3/{account}/websites`
         "GroupId": "GROUPID123",
         "GroupName": "foobar.bulldogs.cloud-BktAdmGrp",
         "Path": "/"
+    },
```
Comment (tenyo) [inline README.md]: Don't we need a `WebsiteConfiguration` in the input request?

**Pull request merged**